core/station/impl/src/repositories/user_group.rs

```diff
@@ -69,34 +82,26 @@ impl Repository<UUID, UserGroup> for UserGroupRepository {
                         .iter()
                         .for_each(|metric| metric.borrow_mut().sub(prev))
                 });
-            }
 
-            self.name_index
-                .refresh_index_on_modification(RefreshIndexMode::CleanupValue {
-                    current: prev.clone().map(|prev| prev.to_index_by_name()),
-                });
+                self.remove_entry_indexes(prev);
+            }
 
             prev
         })
     }
-
-    fn len(&self) -> usize {
-        DB.with(|m| m.borrow().len()) as usize
-    }
 }
 
 impl UserGroupRepository {
-    pub fn find_by_name(&self, name: &str) -> Option<UserGroup> {
-        let user_group_ids = self
-            .name_index
-            .find_by_criteria(UserGroupNameIndexCriteria {
-                name: name.to_string(),
-            });
+    /// Builds the cache from the stable memory repository.
+    ///
+    /// This method should only be called during init or upgrade hooks to ensure that the cache is
+    /// up-to-date with the repository and that we have enough instructions to rebuild the cache.
+    pub fn build_cache(&self) {}
```

Comment:
is this empty on purpose?

Reply:
Nope, nice catch! Just added but the benchmarks showed no improvements based on it, but since this cache is small, we can keep it and have things consistent imo.